provider/pkg/schema/extension_resource.go

```diff
@@ -0,0 +1,272 @@
+package schema
```

Comment:
My aim was to have a single file with a whole resource defined in it - schema & provider implementation. This means that, because of our existing module structure, that `schema` is the only place this can currently sit to not cause import cycles. I also wanted this in the `custom` module, but that would require extracting the core metadata types into a new module at the very least. Can do this as a follow-up maybe.

Reply:
> I also wanted this in the custom module, but that would require extracting the core metadata types into a new module at the very least. 

Then we should do so. The current status quo looks pretty bad to me, I'm not comfortable proceeding with it.